### PR TITLE
Psr-6 auto instrumentation

### PR DESCRIPTION
**Pull request by weslenteche**
Fixes: https://github.com/open-telemetry/opentelemetry-php/issues/1190

### CODE REVIEW

**Comment by bobstrecansky** [comment]
Looks like you had 2 simple Psalm errors to fix: https://github.com/open-telemetry/opentelemetry-php-contrib/actions/runs/8698054664/job/23854436040?pr=256#step:11:18

File: src/Instrumentation/Psr6/composer.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  "require-dev": {
+    "friendsofphp/php-cs-fixer": "^3",
+    "phan/phan": "^5.0",
+    "phpstan/phpstan-mockery": "^1.1.0",
+    "phpstan/phpstan": "^1.1",
+    "phpstan/phpstan-phpunit": "^1.0",
+    "psalm/plugin-phpunit": "^0.16",
+    "open-telemetry/sdk": "^1.0",
+    "phpunit/phpunit": "^9.5",
+    "vimeo/psalm": "^4.0"
```
Comment (brettmc) [inline src/Instrumentation/Psr6/composer.json:27]: try updating this to v5 to fix the failure on 8.3

**Comment by weslenteche** [comment]
@brettmc could you review?

I used the same attributes of the PSR-16 auto instrumentation in the implementation. What do you think, do you have any suggestions for additional attributes?

File: src/Instrumentation/Psr6/composer.json (hunk 1 above)
Comment (weslenteche) [reply src/Instrumentation/Psr6/composer.json:27]: Ok

File: src/Instrumentation/Psr6/src/Psr6Instrumentation.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ */
+class Psr6Instrumentation
+{
+    public const NAME = 'psr6';
+
+    public static function register(): void
+    {
+        $instrumentation = new CachedInstrumentation(
+            'io.opentelemetry.contrib.php.psr6',
+            InstalledVersions::getVersion('open-telemetry/opentelemetry-auto-psr6')
```
Comment (brettmc) [inline src/Instrumentation/Psr6/src/Psr6Instrumentation.php:31]:
```suggestion
            InstalledVersions::getVersion('open-telemetry/opentelemetry-auto-psr6'),
            'https://opentelemetry.io/schemas/1.25.0',
```
(assuming latest semconv, 1.25.0, is the schema you developed against)

**Review by brettmc — APPROVED** [review]

**Pull request merged**